### ISSUE TITLE
Can not show terminal inputs when break `readline()`

### ISSUE THREAD

**Issue by Koishiiko** [microsoft/vscode-cpptools#10865]
### Environment

- OS and version: Windows 10
- VS Code: 1.77.3
- C/C++ extension: v1.15.2
- OS and version of remote machine (if applicable): Ubuntu 22.04
- GDB / LLDB version: GNU gdb (Ubuntu 12.1-0ubuntu1~22.04) 12.1

### Bug Summary and Steps to Reproduce

Bug Summary: 

When use `readline()` and break(`Shift + F5` or `Ctrl + C`) the program, terminal will not display inputs. (But still can execute commands).

Steps to reproduce:

1. create `echo.c`:
```c
#include <stdio.h>
#include <readline/history.h>
#include <readline/readline.h>

int main() {
  char* line = NULL;
  while (line == NULL || strcmp(line, "q") != 0) {
    line = readline("> ");
    printf("%s\n", line);
  }
  return 0;
}
```
2. Press `F5` to debug program.
3. Press `Shift + F5`  or `Ctrl + C` .
4. `cppdbg` terminal will not show inputs.
5. Inputs will display normally if enter `q`.

### Debugger Configurations

```shell
launch.json: 


{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "(gdb) Launch",
      "type": "cppdbg",
      "request": "launch",
      "program": "${workspaceFolder}/echo.out",
      "cwd": "${workspaceFolder}",
      "externalConsole": false,
      "MIMode": "gdb",
      "preLaunchTask": "gcc"
    }
  ]
}


task.json: 

```json
{
  "version": "2.0.0",
  "tasks": [
    {
      "label": "gcc",
      "type": "shell",
      "command": "gcc echo.c -g -lreadline -oecho.out"
    }
  ]
}
```
```


### Debugger Logs

```shell
Maybe no needs.
```

**Comment by WardenGnaw** [comment]
Are you using a Remote Machine or running under WSL? 

I am able to see the `>` prompt without having to ineteract with the terminal

**Comment by Koishiiko** [comment]
> 

Yes, I'm running the program in remote Ubuntu 22.04. Sorry I can't describe it exactly, you can see this about this: 

![gif](https://user-images.githubusercontent.com/62779530/235025205-a211c696-9bb7-4980-b50d-66b8c53b9b92.gif)

After click `stop` button, all inputs in this terminal can not show but can still be executed.
